superset/common/query_context_processor.py

```diff
@@ -600,7 +600,15 @@ def get_payload(
             set_and_log_cache(
                 cache_manager.cache,
                 cache_key,
-                {"data": self._query_context.cache_values},
+                {
+                    "data": {
+                        # setting form_data into query context cache value as well
+                        # so that it can be used to reconstruct form_data field
+                        # for query context object when reading from cache
+                        "form_data": self._query_context.form_data,
```

Comment:
I have concerns here.. We've discussed this at length previously, and I feel we should always consider `form_data` to be chart specific internal state. IIRC, the original reason that form data is being passed along to the celery task is because the legacy charts require it to function. However, the v1 charts shouldn't need it. I know many fields within `form_data` are sort of standardized, but the whole point of `QueryContext`/`QueryObject` was to standardize the chart data request contract, thus decoupling the viz plugins from the backend. Therefore, I feel any field that is required from `form_data` should ideally be allocated a dedicated property on either the query context or query object.